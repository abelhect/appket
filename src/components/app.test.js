import React from 'react';
import {shallow} from 'enzyme';
// import ReactDOM from 'react-dom';
import App from './app';
import '../setupTests';


describe('All tests for this App component', () => {
  const app = shallow(<App marketappslist={[]}/>);

  //Test#1: (visual) App renders correctly
  it('renders correctly', ()=>{
    expect(app).toMatchSnapshot();
  });
});
