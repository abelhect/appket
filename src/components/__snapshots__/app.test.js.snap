// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All tests for this App component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    marketappslist={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
>
          <Router
                    history={
                              Object {
                                        "action": "REPLACE",
                                        "block": [Function],
                                        "createHref": [Function],
                                        "go": [Function],
                                        "goBack": [Function],
                                        "goForward": [Function],
                                        "length": 1,
                                        "listen": [Function],
                                        "location": Object {
                                          "hash": "",
                                          "key": "excpzb",
                                          "pathname": "/signin",
                                          "search": "",
                                          "state": undefined,
                                        },
                                        "push": [Function],
                                        "replace": [Function],
                                      }
                    }
                    path="/"
          >
                    <div>
                              <Connect(Menu)
                                        logo="Papo's AppKet"
                              />
                              <Route
                                        component={[Function]}
                                        path="/market"
                              />
                              <Route
                                        component={[Function]}
                                        path="/signin"
                              />
                              <Route
                                        component={[Function]}
                                        path="/signup"
                              />
                    </div>
          </Router>
</Provider>,
        <p
          className="footer"
>
          Courtesy of Hector A. Martinez
</p>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Router
            history={
                        Object {
                                    "action": "REPLACE",
                                    "block": [Function],
                                    "createHref": [Function],
                                    "go": [Function],
                                    "goBack": [Function],
                                    "goForward": [Function],
                                    "length": 1,
                                    "listen": [Function],
                                    "location": Object {
                                      "hash": "",
                                      "key": "excpzb",
                                      "pathname": "/signin",
                                      "search": "",
                                      "state": undefined,
                                    },
                                    "push": [Function],
                                    "replace": [Function],
                                  }
            }
            path="/"
>
            <div>
                        <Connect(Menu)
                                    logo="Papo's AppKet"
                        />
                        <Route
                                    component={[Function]}
                                    path="/market"
                        />
                        <Route
                                    component={[Function]}
                                    path="/signin"
                        />
                        <Route
                                    component={[Function]}
                                    path="/signup"
                        />
            </div>
</Router>,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              <Connect(Menu)
                            logo="Papo's AppKet"
              />
              <Route
                            component={[Function]}
                            path="/market"
              />
              <Route
                            component={[Function]}
                            path="/signin"
              />
              <Route
                            component={[Function]}
                            path="/signup"
              />
</div>,
            "history": Object {
              "action": "REPLACE",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "excpzb",
                "pathname": "/signin",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            },
            "path": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Menu)
                  logo="Papo's AppKet"
/>,
                <Route
                  component={[Function]}
                  path="/market"
/>,
                <Route
                  component={[Function]}
                  path="/signin"
/>,
                <Route
                  component={[Function]}
                  path="/signup"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "logo": "Papo's AppKet",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/market",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/signin",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/signup",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Courtesy of Hector A. Martinez",
          "className": "footer",
        },
        "ref": null,
        "rendered": "Courtesy of Hector A. Martinez",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Provider
            store={
                        Object {
                                    "dispatch": [Function],
                                    "getState": [Function],
                                    "replaceReducer": [Function],
                                    "subscribe": [Function],
                                    Symbol(observable): [Function],
                                  }
            }
>
            <Router
                        history={
                                    Object {
                                                "action": "REPLACE",
                                                "block": [Function],
                                                "createHref": [Function],
                                                "go": [Function],
                                                "goBack": [Function],
                                                "goForward": [Function],
                                                "length": 1,
                                                "listen": [Function],
                                                "location": Object {
                                                  "hash": "",
                                                  "key": "excpzb",
                                                  "pathname": "/signin",
                                                  "search": "",
                                                  "state": undefined,
                                                },
                                                "push": [Function],
                                                "replace": [Function],
                                              }
                        }
                        path="/"
            >
                        <div>
                                    <Connect(Menu)
                                                logo="Papo's AppKet"
                                    />
                                    <Route
                                                component={[Function]}
                                                path="/market"
                                    />
                                    <Route
                                                component={[Function]}
                                                path="/signin"
                                    />
                                    <Route
                                                component={[Function]}
                                                path="/signup"
                                    />
                        </div>
            </Router>
</Provider>,
          <p
            className="footer"
>
            Courtesy of Hector A. Martinez
</p>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Router
              history={
                            Object {
                                          "action": "REPLACE",
                                          "block": [Function],
                                          "createHref": [Function],
                                          "go": [Function],
                                          "goBack": [Function],
                                          "goForward": [Function],
                                          "length": 1,
                                          "listen": [Function],
                                          "location": Object {
                                            "hash": "",
                                            "key": "excpzb",
                                            "pathname": "/signin",
                                            "search": "",
                                            "state": undefined,
                                          },
                                          "push": [Function],
                                          "replace": [Function],
                                        }
              }
              path="/"
>
              <div>
                            <Connect(Menu)
                                          logo="Papo's AppKet"
                            />
                            <Route
                                          component={[Function]}
                                          path="/market"
                            />
                            <Route
                                          component={[Function]}
                                          path="/signin"
                            />
                            <Route
                                          component={[Function]}
                                          path="/signup"
                            />
              </div>
</Router>,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div>
                <Connect(Menu)
                                logo="Papo's AppKet"
                />
                <Route
                                component={[Function]}
                                path="/market"
                />
                <Route
                                component={[Function]}
                                path="/signin"
                />
                <Route
                                component={[Function]}
                                path="/signup"
                />
</div>,
              "history": Object {
                "action": "REPLACE",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "excpzb",
                  "pathname": "/signin",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              },
              "path": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(Menu)
                    logo="Papo's AppKet"
/>,
                  <Route
                    component={[Function]}
                    path="/market"
/>,
                  <Route
                    component={[Function]}
                    path="/signin"
/>,
                  <Route
                    component={[Function]}
                    path="/signup"
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "logo": "Papo's AppKet",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/market",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/signin",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/signup",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Courtesy of Hector A. Martinez",
            "className": "footer",
          },
          "ref": null,
          "rendered": "Courtesy of Hector A. Martinez",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
